fluent/client/ws/connection.go

```diff
@@ -70,10 +69,18 @@ func NewConnection(conn ext.Conn, opts ConnectionOptions) (Connection, error) {
 		})
 	}
 
-	if opts.ReadHandler != nil {
-		wsc.SetReadHandler(opts.ReadHandler)
+	if opts.ReadHandler == nil {
+		opts.ReadHandler = func(c Connection, _ int, _ []byte, err error) error {
+			if err != nil {
+				_ = c.Close()
```

Comment:
ensures the connection is closed on error, even if the user is not explicitly handling reads